hal/src/nRF52840/usb_hal_cdc.c

```diff
@@ -430,7 +471,7 @@ void usb_hal_attach(void) {
 }
 
 void usb_hal_detach(void) {
-    if (m_usb_instance.state == HAL_USB_STATE_DETACHED) {
+    if (m_usb_instance.state >= HAL_USB_STATE_DETACHED) {
```

Comment:
If the USB device is detached then return, so it should be `m_usb_instance.state <= HAL_USB_STATE_DETACHED `

Reply:
`< HAL_USB_STATE_DETACHED`, right?